src/lib/ui/thermostat/ClimateSettingScheduleCard.tsx

```diff
@@ -0,0 +1,99 @@
+import type { ClimateSettingSchedule } from 'seamapi'
+
+import { formatDateAndTime } from 'lib/dates.js'
+import { ClimateSettingScheduleIcon } from 'lib/icons/ClimateSettingSchedule.js'
+import { useClimateSettingSchedule } from 'lib/seam/thermostats/climate-setting-schedules/use-climate-setting-schedule.js'
+import { ClimateSettingDevice } from 'lib/ui/thermostat/ClimateSettingScheduleDevice.js'
+import { ClimateSettingStatus } from 'lib/ui/thermostat/ClimateSettingStatus.js'
+
+import { DotDivider } from '../layout/DotDivider.js'
+
+interface ClimateSettingScheduleCardProps {
+  climateSettingScheduleId: string
+}
+
+export function ClimateSettingScheduleCard({
+  climateSettingScheduleId,
+}: ClimateSettingScheduleCardProps): JSX.Element {
+  return (
+    <div className='seam-climate-setting-schedule-card'>
+      <Content climateSettingScheduleId={climateSettingScheduleId} />
+    </div>
+  )
+}
+
+function Content(props: {
+  climateSettingScheduleId: string
+}): JSX.Element | null {
+  const { climateSettingScheduleId } = props
+
+  const { climateSettingSchedule } = useClimateSettingSchedule(
+    climateSettingScheduleId
+  )
+
+  if (climateSettingSchedule == null) {
+    return null
+  }
+
+  const name = climateSettingSchedule.name ?? t.fallbackName
+
+  return (
+    <div className='seam-climate-setting-schedule-content'>
+      <div className='seam-climate-setting-schedule-summary-container'>
+        <div className='seam-climate-setting-schedule-icon-block'>
+          <ClimateSettingScheduleIcon />
+        </div>
+        <div className='seam-climate-setting-schedule-summary'>
+          <h5 className='seam-climate-setting-schedule-heading'>{name}</h5>
+          <div className='seam-climate-setting-schedule-subheading'>
+            <ClimateSettingStatus climateSetting={climateSettingSchedule} />
+            <DotDivider />
+            <ClimateSettingScheduleTiming
+              climateSettingSchedule={climateSettingSchedule}
+            />
+          </div>
+        </div>
+      </div>
+      <ClimateSettingDevice deviceId={climateSettingSchedule.device_id} />
+    </div>
+  )
+}
+
+function ClimateSettingScheduleTiming(props: {
+  climateSettingSchedule: ClimateSettingSchedule
+}): JSX.Element | null {
+  const { climateSettingSchedule } = props
+
+  const startTime = new Date(
+    climateSettingSchedule.schedule_starts_at
+  ).getTime()
+
+  const endTime = new Date(climateSettingSchedule.schedule_ends_at).getTime()
+
+  if (Date.now() < startTime)
+    return (
+      <span>
+        {`${t.starts} ${formatDateAndTime(
+          climateSettingSchedule.schedule_starts_at
+        )}`}
+      </span>
+    )
+
+  if (startTime <= Date.now() && Date.now() <= endTime)
```

Comment:
Logic that depends on the current time does not work without [this hook](https://github.com/seamapi/react/blob/main/src/lib/ui/use-interval.ts) or similar since it will only update on render which is not consistent.

Reply:
https://github.com/seamapi/react/pull/378/commits/dd76de64d418e71e13b2152e1c6e37b43a7c91c5#diff-f9bff77895bd5a57c35b2cb284a3804aed48168f365c7b1ceb1f10d72756e9dd